aquia-inc/template-vite-react
# build(dev-deps): update commitlint, semantic-release, and related devDeps

## Summary

Upgraded several devDependencies related to `commitlint` and `semantic-release` to their latest versions. This update ensures the development environment remains up-to-date with the latest improvements, bug fixes, and features for these tools.

### Added

- None

### Changed

- Upgraded `@commitlint/cli` from `^18.6.1` to `^19.4.0`
- Upgraded `@commitlint/config-conventional` from `^18.6.3` to `^19.2.2`
- Upgraded `@semantic-release/commit-analyzer` from `^11.1.0` to `^13.0.0`
- Upgraded `@semantic-release/github` from `^9.2.6` to `^10.1.3`
- Upgraded `@semantic-release/npm` from `^11.0.2` to `^12.0.1`
- Upgraded `@semantic-release/release-notes-generator` from `^12.1.0` to `^14.0.1`
- Upgraded `conventional-changelog-conventionalcommits` from `^7.0.2` to `^8.0.0`
- Upgraded `semantic-release` from `^23.0.2` to `^24.0.0`

### Deprecated

- None

### Removed

- None

### Fixed

- None

## How to test

1. Checkout this branch: `git checkout origin/build/dependencies/semrel`
2. Run `yarn` and ensure all devDependencies are correctly installed without duplication and that there are no issues related to missing or incompatible devDependencies.
3. Run `yarn test` and verify that all tests run and pass.
4. Run `yarn dev` and verify that the application runs as expected.
5. Run `yarn build` and verify that the application builds as expected.
6. Run `yarn build-storybook` and verify the Storybook build succeeds.
7. Verify that the semantic release process works as expected by running a dry-run.
